Core/Src/segment.c

```diff
@@ -63,47 +109,185 @@ void discard_neutrals(void);
 void pull_chip_configuration(void);
 int16_t calc_average(void);
 int8_t calc_therm_standard_dev(int16_t avg_temp);
+void init_chip(cell_asic *chip);
+void set_cell_discharge(cell_asic *chip, uint8_t cell, bool discharge);
+
+/**
+ * @brief Set a bit in a uint16
+ *  
+ * @param number uint16 to change.
+ * @param n Nth bit to change.
+ * @param x true sets, false clears.
+ * @return uint16_t New uint16.
+ */
+inline uint16_t set_uint16_bit(uint16_t number, uint16_t n, bool x)
+{
+	return (number & ~((uint16_t)1 << n)) | ((uint16_t)x << n);
+}
 
-void push_chip_configuration()
+void adbms_wake()
 {
-	LTC6804_wrcfg(ltc68041, NUM_CHIPS, local_config);
+	adBmsCsLow();
+	adBmsCsHigh();
 }
 
-void segment_init()
+/**
+ * @brief Initialize a chip with default values. 
+ * 
+ * @param chip Pointer to chip to initialize.
+ */
+void init_chip(cell_asic *chip)
 {
-	printf("Initializing Segments...");
+	chip->tx_cfga.refon = PWR_UP;
+	chip->tx_cfga.cth = CVT_8_1mV;
+	chip->tx_cfga.flag_d = 0;
 
-	ltc68041 = malloc(sizeof(ltc_config));
-	LTC6804_initialize(ltc68041, &hspi1, GPIOA, SPI_1_CS_Pin);
+	// No soak on AUX ADCs
+	chip->tx_cfga.soakon = SOAKON_CLR;
 
-	pull_chip_configuration();
+	// short soak time by default
+	chip->tx_cfga.owrng = TIME_32US_TO_4_1MS;
 
-	for (int c = 0; c < NUM_CHIPS; c++) {
-		local_config[c][0] = 0xF8;
-		local_config[c][1] = 0x19; /* VUV = 0x619 = 1561 -> 2.4992V */
-		local_config[c][2] = 0x06; /* VOV = 0xA60 = 2656 -> 4.2496V */
-		local_config[c][3] = 0xA6;
-		local_config[c][4] = 0x00;
-		local_config[c][5] = 0x00;
-	}
-	push_chip_configuration();
+	chip->tx_cfga.owa = OWA0;
 
-	start_timer(&voltage_reading_timer, VOLTAGE_WAIT_TIME);
-	start_timer(&therm_timer, THERM_WAIT_TIME);
+	// All GPIOs are inputs by default
+	chip->tx_cfga.gpo = ALL_GPIOS_ARE_INPUTS;
 
-	uint8_t i2c_write_data[NUM_CHIPS][3];
+	// Registers are unfrozen
+	chip->tx_cfga.snap = SNAP_OFF;
+
+	// Charging is deactivated
+	chip->tx_cfga.mute_st = MUTE_ACTIVATED_DISCHARGE_DISABLED;
+
+	// Not an endpoint in the daisy chain
+	chip->tx_cfga.comm_bk = false;
+
+	// IIR filter disabled
+	chip->tx_cfga.fc = IIR_FPA_OFF;
```

Comment:
dont we want this on?

Reply:
We don't have a plan to use it yet so i just kept it off to keep things simple